bootstrap/internal/controllers/rke2config_controller.go

```diff
@@ -225,25 +240,329 @@ func (r *Rke2ConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
-}
+func (r *RKE2ConfigReconciler) handleClusterNotInitialized(ctx context.Context, scope *Scope) (res ctrl.Result, reterr error) {
+
+	// initialize the DataSecretAvailableCondition if missing.
+	// this is required in order to avoid the condition's LastTransitionTime to flicker in case of errors surfacing
+	// using the DataSecretGeneratedFailedReason
+	// if conditions.GetReason(scope.Config, bootstrapv1.DataSecretAvailableCondition) != bootstrapv1.DataSecretGenerationFailedReason {
+	// 	conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, clusterv1.WaitingForControlPlaneAvailableReason, clusterv1.ConditionSeverityInfo, "")
+	// }
+	if !scope.HasControlPlaneOwner {
+		scope.Logger.Info("Requeuing because this machine is not a Control Plane machine")
+		return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+	}
+
+	if !r.RKE2InitLock.Lock(ctx, scope.Cluster, scope.Machine) {
+		scope.Logger.Info("A control plane is already being initialized, requeuing until control plane is ready")
+		return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+	}
+
+	defer func() {
+		if reterr != nil {
+			if !r.RKE2InitLock.Unlock(ctx, scope.Cluster) {
+				reterr = kerrors.NewAggregate([]error{reterr, errors.New("failed to unlock the rke2 init lock")})
+			}
+		}
+	}()
+
+	certificates := secret.NewCertificatesForInitialControlPlane()
+	err := certificates.LookupOrGenerate(
+		ctx,
+		r.Client,
+		util.ObjectKey(scope.Cluster),
+		*metav1.NewControllerRef(scope.Config, bootstrapv1.GroupVersion.WithKind("RKE2Config")),
+	)
+	if err != nil {
+		conditions.MarkFalse(scope.Config, bootstrapv1.CertificatesAvailableCondition, bootstrapv1.CertificatesGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
+		return ctrl.Result{}, err
+	}
+	conditions.MarkTrue(scope.Config, bootstrapv1.CertificatesAvailableCondition)
+
+	token, err := r.generateAndStoreToken(ctx, scope)
+	if err != nil {
+		scope.Logger.Error(err, "unable to generate and store an RKE2 server token")
+		return ctrl.Result{}, err
+	}
+	scope.Logger.Info("RKE2 server token generated and stored in Secret!")
+
+	configStruct, files, err := rke2.GenerateInitControlPlaneConfig(
+		rke2.RKE2ServerConfigOpts{
+			ControlPlaneEndpoint: scope.Cluster.Spec.ControlPlaneEndpoint.Host,
+			Token:                token,
+			ServerURL:            "https://" + scope.Cluster.Spec.ControlPlaneEndpoint.Host + ":9345",
+			ServerConfig:         scope.ControlPlane.Spec.ServerConfig,
+			AgentConfig:          scope.Config.Spec.AgentConfig,
+			Ctx:                  ctx,
+			Client:               r.Client,
+		})
+
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	b, err := kubeyaml.Marshal(configStruct)
+	if err != nil {
+
+		return ctrl.Result{}, err
+	}
+	scope.Logger.Info("Server config marshalled successfully")
+
+	initConfigFile := bootstrapv1.File{
+		Path:        rke2.DefaultRKE2ConfigLocation,
+		Content:     string(b),
+		Owner:       "root:root",
```

Comment:
The values being assigned to Owner & Permissions would be good as constants.

Reply:
Done!